CDCgov/prime-simplereport
# disable hiv bulk upload if feature flag isn't on

# BACKEND PULL REQUEST

## Related Issue

- Next part of #7207 that has the bulk of the upload validation logic
- See #7299 for part 1 
- One more smaller PR to add in frontend device code lookup filtering for inactive diseases as called out in the ticket

## Changes Proposed

- Checks bulk upload submissions to see if a result includes a reference to a LOINC that maps to HIV
     - If yes, checks to see if the upload flag is on and throws an error if yes
- Updates frontend guidance to inform users about this error (see [this slack thread](https://skylight-hq.slack.com/archives/C065V4DHBQF/p1707839947847009) for more context)
- Updates / refactors some tests / tangential objects  

## Additional Information

- Code coverage is a couple of percentage points off, but can't think of anywhere to add a useful test. If anyone has an issue with this, lmk and I can rethink it

## Testing

- Deployed on dev4 (with the HIV flag turned off)
- Add an HIV device and try a bulk upload pointing to that device
- Notice validation error / guidance
- Notice that a regular invalid upload still points to the invalid combination guidance
- Notice that turning the feature flag on results in the file passing through

![Screenshot 2024-02-21 at 10 59 54 AM](https://github.com/CDCgov/prime-simplereport/assets/29645040/5586fd4a-0d86-45e6-adcd-b9b5338a92c2)

added an HIV device for testing / you can use this CSV to get the above

```
patient_id,patient_last_name,patient_first_name,patient_middle_name,patient_street,patient_street2,patient_city,patient_state,patient_zip_code,patient_county,patient_phone_number,patient_dob,patient_gender,patient_race,patient_ethnicity,patient_preferred_language,patient_email,accession_number,equipment_model_name,test_performed_code,test_result,order_test_date,specimen_collection_date,testing_lab_specimen_received_date,test_result_date,date_result_released,specimen_type,ordering_provider_id,ordering_provider_last_name,ordering_provider_first_name,ordering_provider_middle_name,ordering_provider_street,ordering_provider_street2,ordering_provider_city,ordering_provider_state,ordering_provider_zip_code,ordering_provider_phone_number,testing_lab_clia,testing_lab_name,testing_lab_street,testing_lab_street2,testing_lab_city,testing_lab_state,testing_lab_zip_code,testing_lab_phone_number,pregnant,employed_in_healthcare,symptomatic_for_disease,illness_onset_date,resident_congregate_setting,residence_type,hospitalized,icu,ordering_facility_name,ordering_facility_street,ordering_facility_street2,ordering_facility_city,ordering_facility_state,ordering_facility_zip_code,ordering_facility_phone_number,comment,test_result_status
1234,Doe,Jane,,123 Main St,,Birmingham,AL,35226,Jefferson,205-999-2800,1/20/1962,F,White,Not Hispanic or Latino,,,123,hiv device*,94558-4,Detected,12/20/2021 14:00,12/20/2021 14:00,12/20/2021 14:00,12/20/2021 14:00,12/20/2021 14:00,Nasal Swab,1013012657,Smith MD,John,,400 Main Street,,Birmingham,AL,35228,205-888-2000,01D1058442,My Urgent Care,400 Main Street,,Birmingham,AL,35228,205-888-2000,N,N,N,,N,,N,N,My Urgent Care,400 Main Street,Suite 100,Birmingham,AL,35228,205-888-2000,Test Comment,
```



---
